global_helpers/panther_base_helpers.py

```diff
@@ -494,3 +495,26 @@ def m365_alert_context(event):
 def defang_ioc(ioc):
     """return defanged IOC from 1.1.1.1 to 1[.]1[.]1[.]1"""
     return ioc.replace(".", "[.]")
+
+
+def unmock(func: callable, *args, **kwargs):
+    """Wraps mocked functions to perform automatic JSON unmarshalling on the returned value during
+    unit tests, and return the unmodified value during actual log analysis.
+
+    Usage:
+        func: the mocked function you want to call
+        *args: the positional arguments of 'func', if any
+        **kwargs: the keyword arguments of 'func', if any
+
+    Example:
+        To wrap `get_counter("cache-key", force_ttl=True)`, call:
+            unmock(get_counter, "cache-key", force_ttl=True)
+    """
+    # Call the function and save the return value
+    val = func(*args, **kwargs)
+
+    # If this is a mocked function, unmarhsal the returned string as JSON
+    if isinstance(func, MagicMock):
+        return json.loads(val)
```

Comment:
Do you think using a `try` here would be advisable with something like:
```py
if isinstance(func, MagicMock):
	try:
		unmarshal = json.loads(val)
		return unmarshal
	except (TypeError, OverflowError):
		return val
return val
```
Passively returning `val` if an exception is thrown would be safer than the outright `return`. Curious to hear your thoughts, though.

Reply:
I think this is better. I wouldn't wanna do this for _any_ exception, but type errors or overflows seems fine to me!